orchagent/orchdaemon.cpp

```diff
@@ -760,6 +765,15 @@ void OrchDaemon::start()
             continue;
         }
 
+        // output heart beat message to SYSLOG
+        diff = std::chrono::duration_cast<std::chrono::milliseconds>(tend - heart_beat);
+        if (diff.count() >= HEART_BEAT_INTERVAL_MSECS)
+        {
+            heart_beat = tend;
+            // output heart beat message to supervisord with 'PROCESS_COMMUNICATION_STDOUT' event: http://supervisord.org/events.html
+            cout << "<!--XSUPERVISOR:BEGIN-->heartbeat<!--XSUPERVISOR:END-->" << endl;
```

Comment:
>heartbeat [](http://example.com/codeflow?start=45&length=9)

Is this message general enough for future? Assume multiple processes inside one container want to send heartbeats, but the heartbeats should be tracked per process.

Reply:
Yes, when supervisord receive this message, supervisortd will add header information to the message, include process name.

In my drafter PR the watchdog already support new process by check the heartbeat message process name: https://github.com/sonic-net/sonic-buildimage/pull/14686